### PR TITLE
Optimize simulation of flow refresh

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/flows/views.py [hunk 1]
```diff
@@ -1380,12 +1380,20 @@ def post(self, request, *args, **kwargs):
...
+
+                channel_logs = ChannelLog.objects.filter(Q(msg__in=msgs) | Q(connection__in=calls) | Q(connection__in=ussd_sessions))
+
+                channel_logs.delete()
+                msgs.delete()
+                calls.delete()
+                ussd_sessions.delete()
+
                 steps.delete()
+                FlowRun.objects.filter(contact=test_contact).delete()
```
Comment (norkans7) [inline temba/flows/views.py]: This force the query to use the flows_flowrun_contact_id index

File: temba/flows/views.py [hunk 2]
```diff
@@ -1380,12 +1380,20 @@ def post(self, request, *args, **kwargs):
...
-                Msg.objects.filter(contact=test_contact).delete()
-                IVRCall.objects.filter(contact=test_contact).delete()
-                USSDSession.objects.filter(contact=test_contact).delete()
 
-                runs.delete()
+                msgs = Msg.objects.filter(contact=test_contact)
+                calls = IVRCall.objects.filter(contact=test_contact)
+                ussd_sessions = USSDSession.objects.filter(contact=test_contact)
+
+                channel_logs = ChannelLog.objects.filter(Q(msg__in=msgs) | Q(connection__in=calls) | Q(connection__in=ussd_sessions))
```
Comment (nicpottier) [inline temba/flows/views.py]: Are there ever channel logs for simulations?

Comment (norkans7) [reply temba/flows/views.py]: I saw 

0.001 -- DELETE FROM "channels_channellog" WHERE "channels_channellog"."msg_id" IN (17609020, 17609021, 17609022, 17609023, 17609024, 17609025, 17609026, 17609027, 17609028, 17609029, 17609030, 17609031, 17609032, 17609033, 17609034, 17609035, 17609036, 17609037, 17609038, 17609039, 17609040, 17609041, 17609042, 17609043, 17609044, 17609045, 17609046, 17609047, 17609048, 17609049, 17609050, 17609051, 17609052, 17609053, 17609054, 17609055, 17609056, 17609057, 17609058, 17609059, 17609060, 17609061, 17609062, 17609063, 17609064, 17609065, 17609066, 17609067, 17609068, 17609069, 17609070, 17609071, 17609072, 17609073, 17609074, 17609075, 17609076, 17609077, 17609078, 17609079, 17609080, 17609081, 17609082, 17609083, 17609084, 17609085, 17609086, 17609087, 17609088, 17609089, 17609090, 17609091, 17609092, 17609093, 17609094, 17609095, 17609096, 17609097, 17609098, 17609099, 17609100, 17609101, 17609102, 17609103, 17609104, 17609105, 17609106, 17609107, 17609108, 17609109, 17609110, 17609111, 17609112, 17609113, 17609114, 17609115, 17609116, 17609117, 17609118, 17609119, 17609120, 17609121, 17609122, 17609123, 17609124, 17609125, 17609126, 17609127, 17609128, 17609129, 17609130, 17609131, 17609132, 17609133, 17609134, 17609135, 17609136, 17609137, 17609138, 17609139, 17609140, 17609141, 17609142, 17609143, 17609144, 17609145, 17609146)

Comment (norkans7) [reply temba/flows/views.py]: Tho queryset is empty for channel_logs

Comment (nicpottier) [reply temba/flows/views.py]: Ya since we never send nor receive in the simulator I don't think there are any logs to remove here.

Comment (norkans7) [reply temba/flows/views.py]: yep

**Pull request closed without merging**